### DISCUSSION TITLE
Migration with multiple containers running starting at the same time

### DISCUSSION THREAD

**Discussion by farminf** [prisma/prisma#3759] [Q&A] [answered]
We use Prisma in our Nestjs application and the procedure that we have for starting up is that every time we run migrations by `prisma migrate up --experimental`.
In our production, we use AWS ECS for running our application and there we have multiple containers running. we got an error in there that:
```
Error: There are more migrations in the database than locally. This must not happen. Local migration ids: 20200713110004-xxx. Remote migration ids: 20200713110004-xxx, 20200713110004-xxx
```
we have only one migration in our migration file and on the db when I check _migration table, there are 2 identic records!
the only reason that I think is that we run up multiple containers at the same time, and they want to do the migration at the same time, so we get 2 records on the db and then next deployment will fail because it gets out of sync...

anyone have the same experience or have any idea about this? how can we fix this?

**Comment by ryands17** [comment]
@farminf 👋 

As you're using multiple containers with ECS, one option would be to setup a pre-deploy command either with Codebuild/CodeDeploy that performs the migrations steps and updates your ECS containers. This would prevent running the migrations inside containers which is not recommended and also prevent any sync issues with the DB.

So my suggestion would be moving the migrate step outside of ECS before you deploy.

**Reply by farminf** [reply, marked as answer]
Hey @ryands17 :) thanks for the reply. that's an interesting approach... but I ask this cause I knew Sequelize supports this and I was wondering if Prisma would do too.
BTW, I opened an issue about this and got feedback. It seems that we can expect this to be resolved when migration wouldn't be experimental anymore. https://github.com/prisma/migrate/issues/598

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
